force-app/main/default/classes/ServiceService_TEST.cls

```diff
@@ -74,4 +74,58 @@ public with sharing class ServiceService_TEST {
             contactId
         );
     }
+
+    @IsTest
+    private static void testGetServicesByProgramEngagementId() {
+        Id programEngagementId = TestUtil.mockId(ProgramEngagement__c.SObjectType);
+
+        Program__c program1 = new Program__c();
+        program1.Id = TestUtil.mockId(Program__c.SObjectType);
+        program1.Name = 'Program 1';
+        program1.Status__c = 'Active';
+        program1.StartDate__c = Date.today();
+        program1.EndDate__c = Date.today().addDays(30);
+
+        ProgramEngagement__c engagement1 = new ProgramEngagement__c();
+        engagement1.Name = 'Engagement 1';
+        engagement1.Stage__c = 'Enrolled';
+        engagement1.Program__c = program1.Id;
+        engagement1.Role__c = 'Client';
+
+        Service__c service1 = new Service__c();
+        service1.Id = TestUtil.mockId(Service__c.SObjectType);
+        service1.Name = 'Service 1';
+        service1.Program__c = program1.Id;
+        service1.Status__c = 'Active';
+        service1.UnitOfMeasurement__c = 'Hours';
+
+        List<Service__c> services = new List<Service__c>{ service1 };
+        List<Object> expected = new List<Object>{ 'label' };
+
+        BasicStub serviceSelectorStub = new BasicStub(ServiceSelector.class)
+            .withReturnValue('getServicesByProgramEngagementId', Id.class, services);
```

Comment:
Our best practice pattern thus far has been to declare the stubbed method name as a `final String` at the beginning of the test so we can be sure it's the same string in the stub and in the `assertCalledWith` at the end. Not critical if there are no other changes happening.